### PR TITLE
Add coffee

### PR DESCRIPTION
**Pull request by snurby7**
### Change Notes
- A generic form has been added for adding a Coffee
- Server has been updated to add a coffee correctly
- Additional interfaces for later usages.
- Reach Router has been brought into the fold
- README.md has been updated to show the new connection string format (mine was wrong)

### CODE REVIEW

File: backend/contracts/ICoffeeProfile.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/**
+ * @description Interface used to ensure all new coffees following a concrete pattern.
```
Comment (jcuadrosj) [inline backend/contracts/ICoffeeProfile.ts]: Just wondering if there is any way to share contracts between this and the client to not have to upkeep two?

File: client/src/components/coffee/AddCoffeeForm.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+const StyledInputButton = styled.input`
+  border-radius: 4px;
+  border-style: outset;
+`;
+
+type AddCoffeeFormProps = {
+  userId: string;
+};
+
+const AddCoffeeForm = ({ userId }: AddCoffeeFormProps) => {
```
Comment (jcuadrosj) [inline client/src/components/coffee/AddCoffeeForm.tsx]: Is there no validation on the input?

File: backend/contracts/index.ts [hunk 3]
```diff
@@ -1 +1,3 @@
-export * from './user.interface';
\ No newline at end of file
+export * from './ICoffeeProfile';
+export * from './IRequest';
+export * from './IUser';
```
Comment (jcuadrosj) [inline backend/contracts/index.ts]: I like this naming convention instead of .interface!

File: backend/contracts/ICoffeeProfile.ts (hunk 1 above)
Comment (snurby7) [reply backend/contracts/ICoffeeProfile.ts]: yeah, I felt dirty doing that as well. We could use something like this https://yarnpkg.com/en/docs/workspaces to make a package that just exports out common contracts. Created #27 for this.

File: client/src/components/coffee/AddCoffeeForm.tsx (hunk 2 above)
Comment (snurby7) [reply client/src/components/coffee/AddCoffeeForm.tsx]: Not yet, I just wanted to get it working first, another ticket coming. #28

File: backend/contracts/index.ts (hunk 3 above)
Comment (snurby7) [reply backend/contracts/index.ts]: `.interface` is an Angular convention I believe. It's probably in there docs or something lol

File: client/src/util/ApiUtility.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import urlFormatter from 'url';
+
+import { UrlFormatUtility } from './UrlFormatUtility';
+
+export const ApiUtility = {
+  /**
+   * @description A wrapper around the fetch operation in case my needs ever hange
```
[the remark this answers is not in the document]
Comment (snurby7) [reply client/src/util/ApiUtility.ts]: ugh. I'll fix that.

File: client/src/components/coffee/AddCoffeeForm.tsx (hunk 2 above)
Comment (jcuadrosj) [reply client/src/components/coffee/AddCoffeeForm.tsx]: Makes sense.

File: backend/contracts/ICoffeeProfile.ts (hunk 1 above)
Comment (jcuadrosj) [reply backend/contracts/ICoffeeProfile.ts]: Sounds fancy. I like it.

**Pull request merged**